service/api.go

```diff
@@ -130,19 +130,58 @@ func (as *APIServer) FetchTask(w http.ResponseWriter, r *http.Request) {
 	gimlet.WriteJSON(w, t)
 }
 
-// fetchProjectRef returns a project ref given the project identifier
-func (as *APIServer) fetchProjectRef(w http.ResponseWriter, r *http.Request) {
-	id := gimlet.GetVars(r)["identifier"]
-	projectRef, err := model.FindMergedProjectRef(id, "", true)
+// fetchLimitedProjectRef returns a limited project ref given the project identifier
+func (as *APIServer) fetchLimitedProjectRef(w http.ResponseWriter, r *http.Request) {
```

Comment:
Can we add a basic unit test that this route redacts some of the other project ref information?

Reply:
There is no unit test or a good place to add one, and give that it's deprecated it doesn't feel worth going through the setup. I tested it at http://localhost:9090/api/ref/evergreen though and made sure it looks okay 
<img width="368" alt="image" src="https://github.com/evergreen-ci/evergreen/assets/13104717/ae569664-1045-4ad2-9c35-9423bd1b804c">

Reply:
It seems like there are [some existing tests](https://github.com/evergreen-ci/evergreen/blob/main/service/api_patch_test.go) that you can adapt already in this package.

I think we ought to test this because it is a security/permissions concern and we ought to preserve the anti-functionality that this really doesn't return extra fields beyond what it returns today. It doesn't seem like we plan to delete this route ever, so we ought to test/verify that it returns no more than it needs today and won't try to do it in the future.

Reply:
done